src/index.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import MxHeader from './components/MxHeader';
+import MxFooter from './components/MxFooter';
+
+import observe from './utils/observe';
+
+import './style/MxHeader.scss';
+import './style/MxFooter.scss';
+
+let header: Element;
+let footer: Element;
+
+const mount = (className: string, Component: React.ComponentType<any>) => {
+    const element = document.getElementsByClassName(className)[0];
+    if (element) ReactDOM.render(<Component />, element);
```

Comment:
If the element wont be available the previous line will throws error

Reply:
And it should throw an error. Meaning that the user didn't provide the className that the header depends on. It's intentional.